examples/archive.html

```diff
@@ -112,7 +112,7 @@ <h3 id="offcanvasRightLabel">Images</h3>
         getUrl = `https://archive.org/details/${input.value}/`;
         showImages(getUrl); 
       }
-      else if(!input.value.includes('https://')) {
+      else if(!input.value.includes('https://') && !input.value.includes('http://')) {
```

Comment:
This looks good! What happens if someone submits `http://archive.org/details/test-collection`?

Reply:
Ah, i see it's then dealt with in the above code?

Reply:
I see it's working in the video, but shouldn't we need to include an extra condition on line 110? Or am I misunderstanding the flow?

Reply:
Aha - ok, so i think we need to add something on line 121 so that if they input `http` only, we change that to `https` - what do you think?

Reply:
I have a doubt, if we only input `http`, What will we fetch after with only `https`. I mean there's no hostname. Please correct me if I misunderstood the question

Reply:
Ah, sorry, i mean if they input `http://archive.org/details/test-collection`, we should automatically change it to `https://archive.org/details/test-collection`. Does that make sense?

Reply:
I mean, i can see it's working in the video! But... i can't follow the logic very easily. Maybe what we need is to leave comments inline with which code addresses which format. like a comment could say:

```js
// handle input formats like archive.org/details/test-collection

...

// handle input formats like http://archive.org/details/test-collection and change http to https
```

That way the way the code works is easier to follow. What do you think?

Reply:
Understood ! Thanks Jeff.

Reply:
> https://archive.org/details/test-collection

Oh! you mean visual change. Okay, right now the change happens in the background (shortly before `http get` is invoked) and not visually reflected to the user. I will work on the visual change.

Reply:
Ah, actually i am most interested in legibility of the code. On what line exactly is the http/https switch achieved? Thank you!

Reply:
@segun-codes, I think Jeff wants you to add comments for every if else statement

Reply:
Okay... I will work on the legibility and visual change too.

Reply:
Here is the revised version of the code, I think it's more legible this time:

```
    function extractKey() {
      let getUrl;

      if(!input.value.includes('archive.org/details/') && !input.value.includes('https://'))
      {
        getUrl = `https://archive.org/details/${input.value}/`;
        showImages(getUrl); 
      }
      else if(!input.value.includes('https://') && !input.value.includes('http://') && input.value.includes('archive.org/details/')) {
        getUrl = `https://${input.value}`;
        showImages(getUrl);   
      }
      else if(input.value.includes('http://')) {
        getUrl = input.value.replace('http:', 'https:');
        input.value = getUrl;
        showImages(getUrl);   
      }
      else
      {
        getUrl = input.value;
        showImages(getUrl);
      }
    }
```

Reply:
> @segun-codes, I think Jeff wants you to add comments for every if else statement

Oh! okay... I think the code revision I just put forward is more readable with no need for comment.

Reply:
Hi @jywarren, here's a new GIF visually illustrating the http/https switch:

![screen-capture (1)](https://user-images.githubusercontent.com/1612359/199609802-1335152c-2abc-4691-88f8-73f6d385a9da.gif)

Reply:
> Aha - ok, so i think we need to add something on line 121 so that if they input `http` only, we change that to `https` - what do you think?

@jywarren, I agree with your idea. A more legible approach has now been used.

Reply:
Hi @jywarren, kindly review this PR. Many thanks!